### PR TITLE
Add Samples Folder README

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
This README file explains what code samples we offer and where these samples are located.

Once we agree on this, we can adopt it to other languages.

### CODE REVIEW

File: samples/README.MD [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`/samples` ```// package-specific samples```**<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/src`<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/tests`<br>
+
+##  `Hello World` Samples
+Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `Azure.Storage.Blobs` package [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/README.md#examples).
+
+## Package Samples and Hot-To Guides
+Each package folder contains a subfolder called `samples` with additional code samples. These samples can be either short programs contained in *.cs files, or more complete how-to guides (code samples and some commentary) contained *.md files. You can find a shortcuts to main how-to guides in the [**How-To Guides List**](#how-to-guide-list) section below.
```
Comment (AlexGhiondea) [inline samples/README.MD]: `contained *.md` -> `contained in *.md`

File: samples/README.MD [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Samples, Snippets, and How-To Guides
+
+Developers like to learn by looking at code, and so the Azure SDK comes with a myriad of code samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.
+
+# Structure of the Repository
```
Comment (kaerm) [inline samples/README.MD]: I would make it H2 ##

File: samples/README.MD [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+The Azure SDK repository is organized in the following folder structure, with the main sample locations highlighted using **bold** font.
+
+`/samples` - this folder<br>
```
Comment (kaerm) [inline samples/README.MD]: I'm assuming this should be bold too then

File: samples/README.MD [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/<package>` ```// e.g. Azure.Storage.Blobs```<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`README.md` ```// package READMEs contain hello world samples```**<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`/samples` ```// package-specific samples```**<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/src`<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/tests`<br>
+
+##  `Hello World` Samples
+Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `Azure.Storage.Blobs` package [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/README.md#examples).
+
+## Package Samples and Hot-To Guides
```
Comment (kaerm) [inline samples/README.MD]: How-To

File: samples/README.MD [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+Each package folder contains a subfolder called `samples` with additional code samples. These samples can be either short programs contained in *.cs files, or more complete how-to guides (code samples and some commentary) contained in *.md files. You can find a shortcuts to main how-to guides in the [**How-To Guides List**](#how-to-guide-list) section below.
+
+## Sample Applications
+Sometimes we want to illustrate how several APIs or even packages work together in a context of a more complete program. For these cases, we created sample applications that you can look at, download, compile, and execute. These application samples are located on 
+[https://code.msdn.microsoft.com/](https://code.msdn.microsoft.com/).
```
Comment (kaerm) [inline samples/README.MD]: should be [https://docs.microsoft.com/samples](https://docs.microsoft.com/samples)

File: samples/README.MD [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+# How-To Guide List
+This section lists **main** how-to guides for the most commonly used APIs and most common scenarios, i.e. this section does not attempt to be a complete directory of guides contained in this repository. 
+
+#### General How-To Guides
+- How to **authenticate** with Azure services (TODO)
+- How to configure, access, and analyze **logging** information (TODO)
+- How to configure **retries** (TODO)
+- How to access **HTTP response details** (TODO)
+
+#### Azure.Security.KeyVault
```
Comment (kaerm) [inline samples/README.MD]: these service specific ones should already exist and we can link them

File: samples/README.MD [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+&nbsp;&nbsp;&nbsp;&nbsp;`README.md` ```// (this file)```<br>
+`/sdk` ```// folder containing sources for all SDK packages```<br>
+&nbsp;&nbsp;&nbsp;&nbsp;`/<service>` ```// e.g. storage```<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/<package>` ```// e.g. Azure.Storage.Blobs```<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`README.md` ```// package READMEs contain hello world samples```**<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`/samples` ```// package-specific samples```**<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/src`<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/tests`<br>
+
+##  `Hello World` Samples
```
Comment (kaerm) [inline samples/README.MD]: I would just title that Getting Started README

File: samples/README.MD [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `Azure.Storage.Blobs` package [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/README.md#examples).
+
+## Package Samples and Hot-To Guides
+Each package folder contains a subfolder called `samples` with additional code samples. These samples can be either short programs contained in *.cs files, or more complete how-to guides (code samples and some commentary) contained in *.md files. You can find a shortcuts to main how-to guides in the [**How-To Guides List**](#how-to-guide-list) section below.
```
Comment (kaerm) [inline samples/README.MD]: Fix: You can find shortcuts

File: samples/README.MD [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+## Sample Applications
+Sometimes we want to illustrate how several APIs or even packages work together in a context of a more complete program. For these cases, we created sample applications that you can look at, download, compile, and execute. These application samples are located on 
+[https://code.msdn.microsoft.com/](https://code.msdn.microsoft.com/).
+
+# How-To Guide List
```
Comment (kaerm) [inline samples/README.MD]: Should be H2 ##

File: samples/README.MD [hunk 10]
```diff
@@ -0,0 +1,44 @@
+# Samples, Snippets, and How-To Guides
+
+Developers like to learn by looking at code, and so the Azure SDK comes with a myriad of code samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.
+
```
Comment (kaerm) [inline samples/README.MD]: I would here quickly list them (and link to corresponding sections below):
- READMEs
- Samples per service
- docs.microsoft.com/samples
- docs.microsoft.com Service pages with QS and Tutorials
- reference documentation

File: samples/README.MD [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+##  `Hello World` Samples
+Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `Azure.Storage.Blobs` package [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/README.md#examples).
+
+## Package Samples and Hot-To Guides
+Each package folder contains a subfolder called `samples` with additional code samples. These samples can be either short programs contained in *.cs files, or more complete how-to guides (code samples and some commentary) contained in *.md files. You can find a shortcuts to main how-to guides in the [**How-To Guides List**](#how-to-guide-list) section below.
+
+## Sample Applications
+Sometimes we want to illustrate how several APIs or even packages work together in a context of a more complete program. For these cases, we created sample applications that you can look at, download, compile, and execute. These application samples are located on 
+[https://code.msdn.microsoft.com/](https://code.msdn.microsoft.com/).
+
```
Comment (kaerm) [inline samples/README.MD]: We should have code snippets in ref docs - just checked with Alex so we should be calling that out

File: samples/README.MD (hunk 2 above)
Comment (KrzysztofCwalina) [reply samples/README.MD]: I will change.

File: samples/README.MD (hunk 3 above)
Comment (KrzysztofCwalina) [reply samples/README.MD]: not now; we don't [yet] have top level samples in this folder.

File: samples/README.MD (hunk 11 above)
Comment (KrzysztofCwalina) [reply samples/README.MD]: Can you point me to a couple examples of such snippets? I tried to find them and could not.

File: samples/README.MD [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+## Structure of the Repository
+The Azure SDK repository is organized in the following folder structure, with the main sample locations highlighted using **bold** font.
+
+`/samples` - this folder<br>
+&nbsp;&nbsp;&nbsp;&nbsp;`README.md` ```// (this file)```<br>
+`/sdk` ```// folder containing sources for all SDK packages```<br>
+&nbsp;&nbsp;&nbsp;&nbsp;`/<service>` ```// e.g. storage```<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/<package>` ```// e.g. Azure.Storage.Blobs```<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`README.md` ```// package READMEs contain hello world samples```**<br>
```
Comment (pakrym) [inline samples/README.MD]: it's not C# you don't have to use `//` for folder descriptions

Comment (pakrym) [reply samples/README.MD]: It looks very busy right now 
image

File: samples/README.MD (hunk 11 above)
Comment (kaerm) [reply samples/README.MD]: I can point you to the GAd ones in a few hours :)

File: samples/README.MD (hunk 12 above)
Comment (KrzysztofCwalina) [reply samples/README.MD]: What do you think I should do?

Comment (pakrym) [reply samples/README.MD]: `/samples` - this folder<br>
&nbsp;&nbsp;&nbsp;&nbsp;`README.md` - (this file)<br>
`/sdk` - folder containing sources for all SDK packages<br>
&nbsp;&nbsp;&nbsp;&nbsp;`/<service>` - e.g. storage<br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/<package>` - e.g. Azure.Storage.Blobs<br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`README.md`** - package READMEs contain hello world samples<br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`/samples`** - package-specific samples</br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/src`<br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/tests`<br>

**Review by tg-msft — APPROVED** [review]
Looks good

File: samples/README.MD [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `Azure.Storage.Blobs` package [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/README.md#examples).
+
+## Package Samples and How-To Guides
+Each package folder contains a subfolder called `samples` with additional code samples. These samples can be either short programs contained in *.cs files, or more complete how-to guides (code samples and some commentary) contained in *.md files. You can find shortcuts to main how-to guides in the [**How-To Guides List**](#how-to-guide-list) section below.
+
+## Sample Applications
+Sometimes we want to illustrate how several APIs or even packages work together in a context of a more complete program. For these cases, we created sample applications that you can look at, download, compile, and execute. These application samples are located on 
+[https://docs.microsoft.com/samples/](https://docs.microsoft.com/samples/).
+
+## How-To Guide List
```
[the remark this answers is not in the document]
Comment (KrzysztofCwalina) [reply samples/README.MD]: There is a section above called "Package Samples and How-To Guides". It would be strange if this one was called "How-To Guides"

**Pull request merged**